### PR TITLE
feat(Miruro): add Presence

### PR DESCRIPTION
**Pull request by Neo1912**
## Description 
add presence for Miruro

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

File: websites/M/Miruro/presence.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
+const presence = new Presence({
+		clientId: "1280494760418738247",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:6]: This is a placeholder comment, please remove.
```suggestion
const enum Assets { 
```

File: websites/M/Miruro/presence.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+presence.on("UpdateData", async () => {
+	const { pathname, href } = document.location,
+		presenceData: PresenceData = {
+			name: "Miruro",
+			details: "Browsing",
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+			buttons: [
+				{ label: "Watch Now", url: href },
+				{ label: "Join Discord", url: "https://discord.com/invite/pjsdJp96mY" }, // Miruro Discord server
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:19]: You can't advertise in the presence like this.

File: websites/M/Miruro/presence.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+			],
+		};
+
+	if (pathname.startsWith("/info")) {
+		const infoTitle = document
+				.querySelector<HTMLElement>("h1.sc-eZxvvn.cywezh")
+				?.childNodes[0]?.textContent.trim(),
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:26]: Childnodes[0] isn;t the best option here.
```suggestion
		const infoTitle = document
				.querySelector("h1.sc-eZxvvn.cywezh")?.firstChild?.textContent.trim(),
```

File: websites/M/Miruro/presence.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "Now Watching";
+			} else {
+				presenceData.smallImageKey = Assets.Pause;
+				if (currentAnimeProgress)
+					presenceData.smallImageText = `Paused | ${currentAnimeProgress}`;
+			}
+		} else {
+			presenceData.smallImageKey = Assets.Logo;
+			presenceData.smallImageText = "miruro.com";
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:76]:
```suggestion
		if (video) {
				presenceData.smallImageKey = !video?.paused ? Assets.Play : Assets.Pause;
				presenceData.smallImageText = !video?.paused ? "Now Watching" : !!currentAnimeProgress ? `Paused | ${currentAnimeProgress}` : "Paused"
		} else {
			presenceData.smallImageKey = Assets.Logo;
			presenceData.smallImageText = "miruro.com";
```

File: websites/M/Miruro/presence.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+				?.textContent?.match(/\d+/)?.[0],
+			currentAnimeCover = document.querySelector<HTMLImageElement>(
+				"a.sc-dqMHui.bSMGmG img.sc-cgxxSg.kiWytg"
+			)?.src,
+			currentTime = document.querySelector<HTMLElement>(
+				'.vds-time[data-type="current"]'
+			)?.textContent,
+			duration = document.querySelector<HTMLElement>(
+				'.vds-time[data-type="duration"]'
+			)?.textContent;
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:53]:
```suggestion
```

File: websites/M/Miruro/presence.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+		let currentAnimeProgress;
+		if (currentTime && duration)
+			currentAnimeProgress = `${currentTime}/${duration}`;
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:57]:
```suggestion
		if (video?.currentTime && video?.duration)
			currentAnimeProgress = `${video?.currentTime}/${video?.duration}`;
```

File: websites/M/Miruro/presence.ts [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+				"a.sc-dqMHui.bSMGmG img.sc-cgxxSg.kiWytg"
+			)?.src,
+			currentTime = document.querySelector<HTMLElement>(
+				'.vds-time[data-type="current"]'
+			)?.textContent,
+			duration = document.querySelector<HTMLElement>(
+				'.vds-time[data-type="duration"]'
+			)?.textContent;
+
+		let currentAnimeProgress;
```
Comment (darkvillager2) [inline websites/M/Miruro/presence.ts:55]:
```suggestion
		let currentAnimeProgress: String;
```

**Review by darkvillager2 — CHANGES_REQUESTED** [review]
Hello, thanks for contributing!

If you have any questions about the requested changes feel free to dm me (Discord ois @Dark_Ville) or to ask in here!

**Comment by darkvillager2** [comment]
Also i don't seen any timestamps getting used is there a reason for that?

File: websites/M/Miruro/metadata.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.10",
```
Comment (darkvillager2) [inline websites/M/Miruro/metadata.json:2]:
```suggestion
	"$schema": "https://schemas.premid.app/metadata/1.11",
	"apiVersion": 1,
```

**Review by darkvillager2 — CHANGES_REQUESTED** [review]

File: websites/M/Miruro/presence.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+					presence.getTimestampsfromMedia(video);
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "Now Watching";
+			} else {
+				presenceData.smallImageKey = Assets.Pause;
+				presenceData.smallImageText = "Paused";
+			}
+		} else {
+			presenceData.smallImageKey = Assets.Logo;
+			presenceData.smallImageText = "miruro.com";
```
Comment (Bas950) [inline websites/M/Miruro/presence.ts:65]:
```suggestion
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
For the rest LGTM

File: websites/M/Miruro/presence.ts [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+	} else if (pathname.startsWith("/watch")) {
+		const video = document.querySelector<HTMLVideoElement>("video"),
+			currentAnimeTitle =
+				document.querySelector<HTMLElement>(".anime-title a")?.textContent,
+			episodeNumber = document
+				.querySelector<HTMLElement>(".ep-number")
+				?.textContent?.match(/\d+/)?.[0],
+			currentAnimeCover = document.querySelector<HTMLImageElement>(
+				"a.sc-dqMHui.bSMGmG img.sc-cgxxSg.kiWytg"
+			)?.src;
```
Comment (theusaf) [inline websites/M/Miruro/presence.ts:44]: If possible, try using query selectors which avoid computer-generated classes or ids.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Miruro/presence.ts [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+					presence.getTimestampsfromMedia(video);
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "Now Watching";
+			} else {
+				presenceData.smallImageKey = Assets.Pause;
+				presenceData.smallImageText = "Paused";
+			}
+		} else presenceData.smallImageKey = Assets.Logo;
+	} else if (pathname.startsWith("/search")) {
+		const searchQuery = new URLSearchParams(window.location.search).get(
```
Comment (theusaf) [inline websites/M/Miruro/presence.ts:65]: Use `document.location` for repository consistency
```suggestion
		const searchQuery = new URLSearchParams(document.location.search).get(
```

**Review by theusaf — CHANGES_REQUESTED** [review]
Other than this, LGTM

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**